File: rosalind/C/fib.c
#include <stdio.h>
#include <stdlib.h>


int *calcRabbit(int n, int k){
    int pairs[40] = {1, 1};
    int *ret = malloc(sizeof(int));

    for(int i = 2; i < n; i++){
        int f1 = pairs[i-1];
        int f2 = pairs[i-2];
        printf("val: %d", f1+f2);
        pairs[i] = f1+f2;        
    }
    ret = pairs[n];
    return ret;
}

int main(int argc, char *argv[]){
    if(argc < 3){
        printf("Usage: ./fib n k\n");
        exit(1);
    }

    char *p;

    int k = strtol(argv[2], &p, 10);
    int n = strtol(argv[1], &p, 10);
    printf("K: %d\n", k);
    printf("N: %d\n", n);

    printf("%d", *calcRabbit(n, k));
}

File: rosalind/python/fib.py
#!/usr/bin/env python

from sys import argv



def calcRabbits(n,k):
    pairs = [1, 1]
    
    for i in range(2,n):
        try:
            f1 = pairs[i-1]
            f2 = pairs[i-2] * 3
            pairs.append((f1+f2))
        except IndexError:
            pass


    return pairs

if __name__ == "__main__":
    try:
        n = int(argv[1])
        k = int(argv[2])
        print(calcRabbits(n,k))
    except (IndexError, ValueError):
        print("Usage: python fib.py <intN> <intK>")


